js/history.js

```diff
@@ -3,17 +3,80 @@ Fliplet.Registry.set('history-stack', (function() {
   var stack = [];
   var currentIndex = 0;
 
+  const columnsInfo = {
+    original: () => stack[0].data[0],
+    reference: [],
+    getReference: () => {
+      if (!columnsInfo.reference.length) {
+        columnsInfo.reference = columnsInfo.original();
+      }
+      return columnsInfo.reference;
+    },
+    current: [],
+    removedSinceReference: [],
+    removeColumns: (index, amount) => {
+      const reference = columnsInfo.getReference();
+      const removed = columnsInfo.reference.slice(index, index + amount);
+
+      const removedSinceReference = removed.filter(column => reference.includes(column));
+      columnsInfo.removedSinceReference = [...new Set([...columnsInfo.removedSinceReference, ...removedSinceReference])].filter(Boolean);
+
+      columnsInfo.reference.splice(index, amount);
+    },
+    moveColumnsInCollection: (collection, columnOriginalIndexes, toIndex) => {
+      const columnsToMove = columnOriginalIndexes.map(index => collection[index]);
+
+      const collectionClone = [...collection];
+      collectionClone.splice(columnOriginalIndexes[0], columnOriginalIndexes.length);
+      collectionClone.splice(toIndex, 0, ...columnsToMove);
+
+      return collectionClone;
+    },
+    moveColumns: (columnOriginalIndexes, toIndex) => {
+      const reference = columnsInfo.getReference();
+      const current = columnsInfo.current;
+
+      columnsInfo.current = columnsInfo.moveColumnsInCollection(current, columnOriginalIndexes, toIndex);
+      columnsInfo.reference = columnsInfo.moveColumnsInCollection(reference, columnOriginalIndexes, toIndex);
+    },
+    getRenamedColumns: () => {
+      const reference = columnsInfo.getReference();
+      const current = columnsInfo.current;
+      const renamedIndexes = reference.reduce((acc, column, index) => {
+        if (column && current[index] && column !== current[index]) {
+          acc.push(index);
+        };
+        return acc;
+      }, []);
+
+      return renamedIndexes.map(index => ({
+        column: reference[index],
+        newColumn: current[index],
+      }));
+    },
+    getCommitPayload: () => ({
+      deleteColumns: columnsInfo.removedSinceReference,
+      renameColumns: columnsInfo.getRenamedColumns(),
+    }),
+    reset: () => {
+      const newReference = stack[currentIndex].data[0];
+      columnsInfo.reference = newReference;
+      columnsInfo.current = newReference;
+      columnsInfo.removedSinceReference = [];
+    }
+  };
+
   function reset() {
     stack = [];
     currentIndex = 0;
   }
 
   // Clone data without losing the ID
   function cloneSpreadsheetData(data) {
-    return _.map(data, function(row) {
+    return _.map(data, function (row) {
```

Comment:
Something's wrong with your IDE setup for sure, and isn't following our linter rules. See https://docs.google.com/document/d/1XPlQHPbqp3rGGTekcC05bKtwKDDjrU2RMJ4VIHLXp64/edit#

```suggestion
    return _.map(data, function(row) {
```